service/src/main/java/org/mskcc/smile/cpt_gateway/service/impl/MessageHandlingServiceImpl.java

```diff
@@ -188,6 +247,43 @@ public void onMessage(Message msg, Object message) {
             }
         });
     }
+
+    private void setupCMOUpdateRequestSubscriber(Gateway gateway,
+                                                 MessageHandlingService messageHandlingService)
+        throws Exception {
+        gateway.subscribe(CMO_UPDATE_REQUEST_TOPIC, Object.class, new MessageConsumer() {
+            public void onMessage(Message msg, Object message) {
+                try {
+                    messageHandlingService.newRequestHandler(
```

Comment:
Request updates and sample updates are going to be handled by the same message handler?

Reply:
Yeah, all the handlers are basically taking the NATS message and posting it to a different CPT endpoint.